include/SDL3/SDL_filesystem.h

```diff
@@ -138,6 +138,54 @@ extern DECLSPEC char *SDLCALL SDL_GetBasePath(void);
  */
 extern DECLSPEC char *SDLCALL SDL_GetPrefPath(const char *org, const char *app);
 
+
+/**
+ * The type of the OS-provided default folder for a specific purpose.
+ *
+ * Note that the Trash folder isn't included here, because trashing files usually
+ * involves extra OS-specific functionality to remember the file's original
+ * location.
+ *
+ * \sa SDL_GetPath
+ */
+typedef enum
+{
+  SDL_FOLDER_HOME,
+  SDL_FOLDER_DESKTOP,
+  SDL_FOLDER_DOCUMENTS,
+  SDL_FOLDER_DOWNLOADS,
+  SDL_FOLDER_MUSIC,
+  SDL_FOLDER_PICTURES,
+  SDL_FOLDER_PUBLICSHARE,
+  SDL_FOLDER_SAVEDGAMES,
+  SDL_FOLDER_SCREENSHOTS,
+  SDL_FOLDER_TEMPLATES,
+  SDL_FOLDER_VIDEOS,
+} SDL_Folder;
```

Comment:
Can you please add comments describing what these folders are?
I think it also makes sense to add enums for the other folders that SDL already provides and remove those functions for SDL 3.0. That can happen as a separate commit after this is merged.

Note that the ABI is not frozen yet, so until SDL 3.0 official release you can reorder these and add values in the middle, which won't be possible after release.

Reply:
It would also be useful to note which platforms support each value.

Reply:
> I think it also makes sense to add enums for the other folders that SDL already provides and remove those functions for SDL 3.0.

I did consider doing this, but I my original thought about SDL_GetPath was that it would provide OS-specific folders for interoperability with standard applications, such as generic media players, cloud/drive software or other. This allowed me to keep things simple and provide a wrapper over just one functionality per platform, in accordance with the "Do one thing and do it well" principle.

Since SDL_Get{Base,Pref}Path are meant to provide a private space for application-specific data, and don't necessarily represent a folder with any particular meaning in the eyes of the OS, and the code between the three functions do not overlap a lot, I thought it would be preferable to keep them apart.

I don't want to advocate any option in particular, but since I had spent time already thinking about it, I thought I might share my reflection.

Reply:
Fair enough, we can leave this as-is for now.

Thanks!

Reply:
For future reference:

`SDL_GetBasePath` could in principle be implemented either here or separately. I would personally agree with what @Semphriss has said about "do one thing, well": the functionality added here is a wrapper around something like the `xdg-user-dirs` spec or `SHGetKnownFolderPath`, whereas `SDL_GetBasePath` goes through a completely different code path that locates the executable in an OS-dependent way, and that feels like different functionality to me.

`SDL_GetPrefPath` *can't* be implemented here, because it requires additional inputs (the organization name and the app name) which are not known to SDL as globals. So I think that one needs to stay separate, regardless.

It would be possible for `SDL_GetPath` to have enum members that return something like XDG `~/.config`, `~/.cache` and `~/.local/share`, and other OSs' equivalents like `C:\Users\bob\AppData\Roaming` - but if it did, they would only be the non-app-specific directories, and the application would still need to know the OS convention for creating its own per-app subdirectories, and that is itself OS-specific. According to `SDL_GetPrefPath()`, the convention is to add the organization name and app name as two levels of hierarchy on Windows, but just use the app name as a single level of hierarchy on macOS and Linux.

Reply:
> SDL_GetPrefPath can't be implemented here, because it requires additional inputs (the organization name and the app name) which are not known to SDL as globals. So I think that one needs to stay separate, regardless.

If the new API had an enum for AppData and its equivalent on various platforms (which I think you're suggesting), then GetPrefPath could at least call into this in its implementation, even if it still has other platform specific code.